main.rb

```diff
@@ -0,0 +1,226 @@
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Style/CaseEquality
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
```

Comment:
We can improve on the customization by``` return to_enum(name_of_method) unless block_given?``` instead of 
``` return to_enum unless block_given?```. 
With this  way 
```[1,2,3].my_each``` will return ```<Enumerator: [1,2,3]:my_each>``` instead of ```<Enumerator: [1,2,3]:each>```
You can replicate this in other method that return enum when no block is giving